### ISSUE TITLE
[Feature] Add error message to clarify table behavior for vector inputs

### ISSUE THREAD

**Issue by francoishamon** [GEOSX/GEOSX#1230]
**What is the requested feature?**
Let's consider a vector quantity such as permeability that we want to provide to GEOSX using tables. I realized that if the user provides two tables for the x and y components, but just sets a uniform value through `FieldSpecification` for the z component, then GEOSX silently ignores the tables for the x and y components and set a uniform value for all components:

To be specific, the following input would cause the problem:
```
    <FieldSpecification
      name="permx"
      component="0"
      initialCondition="1"
      setNames="{ all }"
      objectPath="ElementRegions/Region1/cb1"
      fieldName="permeability"
      scale="1.0"
      functionName="permxFunc"/>

    <FieldSpecification
      name="permy"
      component="1"
      initialCondition="1"
      setNames="{ all }"
      objectPath="ElementRegions/Region1/cb1"
      fieldName="permeability"
      scale="1.0"
      functionName="permyFunc"/>

    <FieldSpecification
      name="permz"
      component="2"
      initialCondition="1"
      setNames="{ all }"
      objectPath="ElementRegions/Region1/cb1"
      fieldName="permeability"
      scale="1.0e-13"/>
``` 

**Is your request related to a specific problem?**
@tang39 's CO2 simulations in heterogeneous permeability fields.

**Describe the solution you'd like**
Throw an error message to let the user know that all the components should be specified with a table (if tables were provided for some components, but not for all). 

**Describe alternatives you've considered**
1) Throw a warning saying that the tables are going to be ignored because a table is missing for one of the components.
2) Do not ignore the tables: set the components of the vector using tables for the components that have a table; use a uniform value for the components that do not have a table.

**Additional context**
The problem can be reproduced on the branch `bugfix/hamon/perforations` using these [files](https://totalcse.box.com/s/0yomzdymlp9kt80f9k91ka0btehubfu7).

@cssherman what do you think?

**Comment by cssherman** [comment]
@francoishamon - these are all applied independently in the order they are specified.  Also, I've double-checked using the sedov model, and the FieldSpecification without a table works as expected when applying to a component of stress.

Looking at the files you attached, it appears that you are applying a scale of 1e-13 to the permx and permy results, and that your tables are in base units (~1e-13):

```xml
<FieldSpecification
      name="permy"
      component="1"
      initialCondition="1"
      setNames="{ all }"
      objectPath="ElementRegions/Region1/cb1"
      fieldName="permeability"
      scale="1.0e-13"
      functionName="permyFunc"/>
```

If this is the case, then the actual permx/permy values you are specifying are on the order of 1e-26.  Could you try re-running with scale=1 for these?

**Comment by francoishamon** [comment]
@cssherman Yes we changed the `scale="1.0e-13"` to `scale="1.0"` to make Hewei's problems work but I realized now that I have not double-checked the permeability magnitude after doing that, so you must be right. That explains the problem. With the initial test files posted on this issue, the permeability magnitude (`sqrt( k_x^2 + k_y^2 + k_z^2 )`) displayed in Visit just appeared homogeneous because these x and y component values were very small, so I got confused. Thanks!
